### PR TITLE
WIP: Add UML diagrams to improve documentation for "Details"

### PR DESCRIPTION
**Pull request by bruderol**
Reverting my reverted merge of #993 ---

can only be merged once we installed graphvis properly to work with our automated documentation build - currently does not yet work.

### CODE REVIEW

File: docs/features/details/Details.md [hunk 1]
```diff
@@ -2,18 +2,38 @@
 
 Details are a generic data structure used for storing additional (application specific) information inside the Scenarioo documentation.
 
-This `details` are simple maps of key-value-pairs. Each `entry` in such a details map has a `key` and a `value`. 
+This `details` are simple maps of key-value-pairs that define further properties of an object. Each such  property which has a `key` and a `value`.
```
Comment (davidlahner) [inline docs/features/details/Details.md]: "Each such  property which has" -> "Each such  property has"

File: docs/features/details/Details.md [hunk 2]
```diff
@@ -2,18 +2,38 @@
...
-The `value` of such an entry can be of different kinds:
+
+This class diagram shows the generic data structure of what you can store in a details map as values:
+
+```puml { src="features/details/class-diagram.puml" }
+```
+
+## Details - Example Object Diagram
+
+Here is such an example of such a data structure you could add like this to the details of a scenario object:
```
Comment (davidlahner) [inline docs/features/details/Details.md]: "Here is an example of such a data structure that you could add to the details of a scenario object:"

File: docs/features/details/Details.md [hunk 3]
```diff
@@ -2,18 +2,38 @@
...
+
+```puml { src="features/details/example.object-diagram.puml" }
+```
+
+See further examples as xml examples further below.
```
Comment (davidlahner) [inline docs/features/details/Details.md]: "See further examples as xml examples further below." -> "See additional examples as xml examples further below."

**Review by davidlahner — CHANGES_REQUESTED** [review]
Thanks for this effort.
I improved the wording in a couple of sentences to make the meaning clearer.
Where and how do we need to install graphvis? On the demo server?

File: docs/features/details/Details.md (hunk 1 above)
Comment (bruderol) [reply docs/features/details/Details.md]: awesome! Thanks!

File: docs/features/details/Details.md (hunk 2 above)
Comment (bruderol) [reply docs/features/details/Details.md]: Yes, I have to improve my writting! Damn good improvement! Great! :-D 👍

**Comment by bruderol** [comment]
@eggerschwiler I changed what you proposed. Thanks! Will open an issue to get the docu build working with the UML diagrams.

**Comment by bruderol** [comment]
@lynparbone I close this - since we opened a new branch with new PR #999 for testing the docu build - once this is fixed there - we can merge it there.

**Pull request closed without merging**